instrumentor/controllers/instrumentedapplication_controller.go

```diff
@@ -201,7 +240,10 @@ func (r *InstrumentedApplicationReconciler) startDetection(ctx context.Context,
 }
 
 func (r *InstrumentedApplicationReconciler) shouldStartDetection(app *v1.InstrumentedApplication) bool {
-	return app.Status.InstrumentationDetection.Phase == v1.PendingInstrumentationDetectionPhase
+	if app.Status.InstrumentationDetection.Phase == v1.PendingInstrumentationDetectionPhase {
```

Comment:
Is there a specific reason for this change?

Reply:
I considered modifying the logic in this function but I didn't do it eventually, so it is indeed redundant, I will revert that